gen/rules/src/main/scala/clue/gen/Generator.scala

```diff
@@ -497,11 +544,17 @@ trait Generator {
     def valName(prefix: String): Pat.Var = Pat.Var(Term.Name(s"$prefix$name"))
 
     val eqDef = Option.when(catsEq)(
-      q"implicit val ${valName("eq")}: cats.Eq[$n] = cats.Eq.fromUniversalEquals"
+      if (jitDecoder)
+        q"implicit val ${valName("eq")}: cats.Eq[$n] = _root_.io.circe.Json.eqJson.asInstanceOf[cats.Eq[$n]]"
```

Comment:
Maybe `summon[Eq[Json]]` in case it changes name?

Reply:
That's pretty much what I did at first 😅 the problem is, the compiler can't distinguish the opaque type from `Json`, so `summon[Eq[Json]`` ends up attempting to circularly summon the very implicit we are defining 😂 

> in case it changes name?

This would be binary-breaking anyway, so unlikely.

Reply:
I see. I guess we could summon it once per file and assign it a name ourselves. But may be overkill.